src/library/scala/collection/immutable/LazyList.scala

```diff
@@ -189,6 +189,33 @@ import scala.runtime.Statics
   *  }
   *  }}}
   *
+  *  - Building a LazyList with `#::` or `#:::` creates the tail lazily.
+  *  The head, the tail and whether the list is empty or not can be initially unknown.
+  *  They are all evaluated at once, but using any of those constructs, evaluation
+  *  of the tail doesn't yet force any of the effects in creating it.
+  *
+  *  Delaying the evaluation of whether a LazyList is empty or not until it's needed
+  *  allows LazyList to not eagerly evaluate any elements on a call to `filter`.
+  *
+  *  Only when it's further evaluated (which may be never!) any of the elements gets
+  *  forced.
+  *
+  *  for example:
+  * 
+  *  {{{
+  *  def tailWithSideEffect: LazyList[Nothing] = {
+  *    println("getting empty LazyList")
+  *    LazyList.empty
+  *  }
+  *
+  *  val emptyTail = tailWithSideEffect //prints "getting empty LazyList"
```

Comment:
nit: I feel like it's generally considered better style to have a space after the `//`